Use python 2.7.13 on the worker nodes

amaltaro opened the pull request:
WMAgent version of #7807
If we decide to go with the new COMP python build, then we need to add this PR to the agent.
However, before that we need to have it deployed on CVMFS and OSG sites.

amaltaro committed:
Use python 2.7.13 on the worker nodes

ericvaandering commented:
Seems to me we can be smarter than having to update version numbers every time. :-) Can we auto discover this?

amaltaro commented:
We can always be smarter, though it depends on the cost-benefit, which for this case is very small :-) 
Anyways, I'll update this patch such that it looks for the latest python release and just use that. That will require special hooks for the moment we migrate to python 3 as well.

amaltaro committed:
Automatically discover the latest python release; adapt the binary to either python2 or python3

Handle weird python versions; check if python dir exist instead of init.sh

rename scram arch var

Use just list

amaltaro commented:
Eric, the second commit has the changes you suggested. I still need to test it though.

amaltaro commented:
Tested both at FNAL and CERN and it works Ok, though I still want to test it in other US and EU sites. @ticoann good for a review though.

ticoann commented on etc/submit.sh:
```diff
@@ -1,4 +1,72 @@
+    exit 7
+  fi
+
+  local latest='2.7.6'         # default release
```
shouldn't we set default to 2.7.13 ?

amaltaro replied:
It doesn't exist in cvmfs yet, that's why I made an existent one as default.

ticoann replied:
I see. Thanks,

ericvaandering commented:
test this please

ericvaandering commented:
(testing Jenkins, not the PR)

ericvaandering commented:
test this please

amaltaro commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Fix #7789 - Use python 2.7.13 on the worker nodes](https://trello.com/c/eBjWafrE/29-fix-7789-use-python-2-7-13-on-the-worker-nodes)

amaltaro commented:
And I need to make sure it can properly handle a "version" directory like `2.7.11-comp`. I'll test it tomorrow.

amaltaro commented:
sigh... I rebased it with master instead of the wmagent branch!

amaltaro commented:
@vkuznet can you point me to the script you are talking about in the meeting?

vkuznet commented:
/cvmfs/cms.cern.ch/common/das_client

On  0, Alan Malta Rodrigues <notifications@github.com> wrote:
> @vkuznet can you point me to the script you are talking about in the meeting?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/dmwm/WMCore/pull/7789#issuecomment-295771261

amaltaro commented:
Thanks, Valentin. Much easier indeed.

amaltaro commented:
Seems to be working fine. I tested it in a couple US sites, Vienna, Bari, Jinr and there were no issues.
I'm merging it now in order to cut a new wmagent tag. However, please feel free to review it.
